src/ImageSharp.Web/Providers/FileProviderImageProvider.cs

```diff
@@ -54,7 +55,7 @@ public virtual bool IsValidRequest(HttpContext context)
         /// <inheritdoc/>
         public Task<IImageResolver> GetAsync(HttpContext context)
         {
-            IFileInfo fileInfo = this.fileProvider.GetFileInfo(context.Request.Path);
+            IFileInfo fileInfo = this.fileProvider.GetFileInfo(WebUtility.UrlDecode(context.Request.Path));
```

Comment:
It's probably enough to get the underlying value from the `PathString`, as it's actually the implicit `ToString()` that causes it to encode the path:
```suggestion
            IFileInfo fileInfo = this.fileProvider.GetFileInfo(context.Request.Path.Value);
```

See also: https://stackoverflow.com/questions/66471763/inconsistent-url-decoding-of-httprequest-path-in-asp-net-core

Reply:
OK sure, the potential issue with space/+ mentioned in your link doesn't seem to be a problem (for me) either.